src/qubx/core/strategy.py

```diff
@@ -389,71 +428,76 @@ def __check_how_to_trigger_and_fit_strategy(
+                _SW.start("strategy.on_event")
+                self.strategy.on_event(self, _strategy_trigger_on_event)
+                self._fails_counter = 0
+            except Exception as strat_error:
+                # - probably we need some cooldown interval after exception to prevent flooding
+                logger.error(
+                    f"[{self.time()}]: Strategy {self.strategy.__class__.__name__} raised an exception: {strat_error}"
+                )
+                logger.opt(colors=False).error(traceback.format_exc())
 
-                # - if strategy still fitting - skip on_event call
-                if self.__fit_is_running:
-                    _SW.stop("StrategyContext._process_incoming_data")
-                    logger.warning(
-                        f"[{self.time()}] {self.strategy.__class__.__name__}::on_event() is SKIPPED for now because is being still fitting !"
-                    )
-                    continue
-
-                try:
-                    _SW.start("strategy.on_event")
-                    self.strategy.on_event(self, _strategy_trigger_on_event)
-                    _fails_counter = 0
-                except Exception as strat_error:
-                    # - probably we need some cooldown interval after exception to prevent flooding
-                    logger.error(
-                        f"[{self.time()}]: Strategy {self.strategy.__class__.__name__} raised an exception: {strat_error}"
-                    )
-                    logger.opt(colors=False).error(traceback.format_exc())
+                #  - we stop execution after let's say maximal number of errors in a row
+                self.__fails_counter += 1
+                if self.__fails_counter >= StrategyContext.MAX_NUMBER_OF_STRATEGY_FAILURES:
+                    logger.error("STRATEGY FAILURES IN THE ROW EXCEEDED MAX ALLOWED NUMBER - STOPPING ...")
+                    return True
+            finally:
+                _SW.stop("strategy.on_event")
 
-                    #  - we stop execution after let's say maximal number of errors in a row
-                    if (_fails_counter := _fails_counter + 1) >= StrategyContext.MAX_NUMBER_OF_STRATEGY_FAILURES:
-                        logger.error("STRATEGY FAILURES IN THE ROW EXCEEDED MAX ALLOWED NUMBER - STOPPING ...")
-                        channel.stop()
-                        break
-                finally:
-                    _SW.stop("strategy.on_event")
+        # - notify poition and portfolio loggers
+        self._logging.notify(self.time())
 
-            # - stop loop latency measurement
-            _SW.stop("StrategyContext._process_incoming_data")
+        return False
+
+    def _process_incoming_data_loop(self, channel: CtrlChannel):
+        logger.info("(StrategyContext) Start processing market data")
+
+        while channel.control.is_set():
+            # - start loop latency measurement
+            _SW.start("StrategyContext._process_incoming_data")
```

Comment:
I noticed you repeated the _SW start process_incoming_data twice. Probably this line can be deleted